lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Migrations.hs

```diff
@@ -196,18 +197,21 @@ spec = describe "SHELLEY_MIGRATIONS" $ do
                         (errMsg404NoWallet $ sourceWallet ^. walletId)
                     ]
 
+
     it "SHELLEY_CREATE_MIGRATION_PLAN_04 - \
         \Cannot create a plan for a wallet that only contains dust."
         $ \ctx -> runResourceT $ do
-            liftIO $ pendingWith
-                "Disabled until a real dust wallet is available."
-            let payloadRestore = Json [json| {
-                    "name": "Dust Shelley Wallet",
-                    "mnemonic_sentence": #{mnemonicToText onlyDustWallet},
-                    "passphrase": #{fixturePassphrase},
-                    "style": "random"
-                    } |]
-            sourceWallet <- unsafeResponse <$> postWallet ctx payloadRestore
+            sourceWallet <- emptyWallet ctx
+            srcAddrs <- map (getApiT . fst . view #id)
+                <$> listAddresses @n ctx sourceWallet
+
+            -- Big enough to mint and sent to sourceWallet, but small enough
+            -- that the migration to targetWallet then fails.
+            let edgeCase = Coin 3_300_000
+
+            liftIO $ _mintSeaHorseAssets ctx 10 edgeCase srcAddrs
+            waitForTxImmutability ctx
+
```

Comment:
Suggestion: add a check to see that the source balance is as we expect it (i.e., that it is greater than zero, and the set of assets in the balance is not the empty set).